File: Games/GameOfGames/src/main/java/jokatu/game/games/gameofgames/views/GameOfGamesViewResolverFactory.java
package jokatu.game.games.gameofgames.views;

import jokatu.game.games.gameofgames.game.GameOfGames;
import jokatu.game.games.gameofgames.player.GameOfGamesPlayer;
import jokatu.game.viewresolver.ViewResolver;
import jokatu.game.viewresolver.ViewResolverFactory;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

@Component
public class GameOfGamesViewResolverFactory extends ViewResolverFactory<GameOfGamesPlayer, GameOfGames> {
	@NotNull
	@Override
	protected Class<GameOfGames> getGameClass() {
		return GameOfGames.class;
	}

	@NotNull
	@Override
	protected ViewResolver<GameOfGamesPlayer, GameOfGames> getResolverFor(GameOfGames castGame) {
		return new GameOfGamesViewResolver(castGame);
	}
}

File: Games/GameOfGames/src/main/kotlin/jokatu/game/games/gameofgames/views/GameOfGamesViewResolverFactory.kt
package jokatu.game.games.gameofgames.views

import jokatu.game.games.gameofgames.game.GameOfGames
import jokatu.game.player.StandardPlayer
import jokatu.game.viewresolver.ViewResolver
import jokatu.game.viewresolver.ViewResolverFactory
import org.springframework.stereotype.Component

@Component
class GameOfGamesViewResolverFactory : ViewResolverFactory<StandardPlayer, GameOfGames>(GameOfGames::class) {
	override fun getResolverFor(castGame: GameOfGames): ViewResolver<StandardPlayer, GameOfGames> {
		return GameOfGamesViewResolver(castGame)
	}
}
